### PR TITLE
docs: Fix `@returns`, `@remarks`, and `@privateRemarks` comments to not include trailing `-`s

### PR DESCRIPTION
**Pull request by Josmithr**
Per TSDoc spec, neither `@returns`, `@remarks`, nor `@privateRemarks` comments should include a `-` between the tag and the corresponding documentation. Only `@param` comments should use `-` syntax.

This PR removes unwanted `-`s in these comments. \+ Some misc. TSDoc improvements.

### CODE REVIEW

File: packages/dds/sequence/src/intervalCollection.ts [hunk 1]
```diff
@@ -636,8 +636,8 @@ export interface IIntervalCollection<TInterval extends ISerializableInterval>
 	 * @param intervalType - type of the interval. All intervals are SlideOnRemove. Intervals may not be Transient.
 	 * @param props - properties of the interval
 	 * @param stickiness - {@link (IntervalStickiness:type)} to apply to the added interval.
-	 * @returns - the created interval
-	 * @remarks - See documentation on {@link SequenceInterval} for comments on interval endpoint semantics: there are subtleties
+	 * @returns the created interval
```
Comment (alexvy86) [inline packages/dds/sequence/src/intervalCollection.ts:639]: Nit for consistency with the other changes like this one

```suggestion
	 * @returns The created interval
```

File: packages/drivers/odsp-driver/src/compactSnapshotParser.ts [hunk 2]
```diff
@@ -215,7 +215,7 @@ function readSnapshotSection(node: NodeTypes) {
 /**
  * Converts snapshot from binary compact representation to tree/blobs/ops.
  * @param buffer - Compact snapshot to be parsed into tree/blobs/ops.
- * @returns - tree, blobs and ops from the snapshot.
+ * @returns tree, blobs and ops from the snapshot.
```
Comment (alexvy86) [inline packages/drivers/odsp-driver/src/compactSnapshotParser.ts:218]: Idem

```suggestion
 * @returns Tree, blobs and ops from the snapshot.
```

File: packages/loader/driver-utils/src/parallelRequests.ts [hunk 3]
```diff
@@ -405,7 +405,7 @@ const waitForOnline = async (): Promise<void> => {
  * @param logger - logger object to use to log progress & errors
  * @param signal - cancelation signal
  * @param scenarioName - reason for fetching ops
- * @returns - an object with resulting ops and cancellation / partial result flags
+ * @returns an object with resulting ops and cancellation / partial result flags
```
Comment (alexvy86) [inline packages/loader/driver-utils/src/parallelRequests.ts:408]: Idem
```suggestion
 * @returns An object with resulting ops and cancellation / partial result flags
```

File: packages/runtime/container-runtime/src/gc/garbageCollection.ts [hunk 4]
```diff
@@ -724,7 +727,7 @@ export class GarbageCollector implements IGarbageCollector {
 	 * This function identifies nodes that were referenced since the last run.
 	 * If these nodes are currently unreferenced, they will be assigned new unreferenced state by the current run.
 	 *
-	 * @returns - a list of all nodes referenced from the last local summary until now.
+	 * @returns a list of all nodes referenced from the last local summary until now.
```
Comment (alexvy86) [inline packages/runtime/container-runtime/src/gc/garbageCollection.ts:730]:
```suggestion
	 * @returns A list of all nodes referenced from the last local summary until now.
```

File: packages/runtime/container-runtime/src/summary/runningSummarizer.ts [hunk 5]
```diff
@@ -519,7 +519,7 @@ export class RunningSummarizer extends TypedEventEmitter<ISummarizerEvents> impl
 	 * @param before - set of instructions to run before running the action.
 	 * @param action - action to perform.
 	 * @param after - set of instructions to run after running the action.
-	 * @returns - result of action.
+	 * @returns result of action.
```
Comment (alexvy86) [inline packages/runtime/container-runtime/src/summary/runningSummarizer.ts:522]:
```suggestion
	 * @returns Result of action.
```

**Review by alexvy86 — APPROVED** [review]
Just nits below, nothing blocking

File: packages/runtime/container-runtime/src/summary/runningSummarizer.ts (hunk 5 above)
Comment (Josmithr) [reply packages/runtime/container-runtime/src/summary/runningSummarizer.ts:522]:
```suggestion
	 * @returns The result of the action.
```

**Pull request merged**